pyproject.toml

```diff
@@ -13,8 +13,108 @@
 # limitations under the License.
 
 [build-system]
-build-backend = "setuptools.build_meta:__legacy__"
+build-backend = "setuptools.build_meta"
 requires = [
-    "setuptools",
+    "setuptools>=64.0.0",
     "cython>=0.29,<0.30",
+    "tomli  ; python_version < '3.11'",
 ]
+
+[project]
+name = "ucx-py"
+dynamic = [
+    "version",
+]
+description = "Python Bindings for the Unified Communication X library (UCX)"
+readme = { file = "README.md", content-type = "text/markdown" }
+authors = [
+    { name = "NVIDIA Corporation" },
+]
+license = { text = "BSD-3-Clause" }
+requires-python = ">=3.6"
+dependencies = [
+    "numpy",
+    "psutil",
+    "pynvml",
+]
+classifiers = [
+    "Intended Audience :: Developers",
+    "Intended Audience :: System Administrators",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: Python",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Topic :: System :: Hardware",
+    "Topic :: System :: Systems Administration",
+    "Programming Language :: Python :: 3",
+]
+
+[project.optional-dependencies]
+test = [
+    "pytest",
+    "pytest-asyncio",
+]
+
+[project.urls]
+Homepage = "https://github.com/rapidsai/ucx-py"
+
+[tool.isort]
+line_length = 79
+multi_line_output = 3
+include_trailing_comma = true
+force_grid_wrap = 0
+combine_as_imports = true
+order_by_type = true
+known_dask = [
+    "dask",
+    "distributed",
+    "dask_cuda",
+]
+known_rapids = [
+    "rmm",
+    "cuml",
+    "cugraph",
+    "dask_cudf",
+    "cudf",
+]
+known_first_party = [
+    "ucp",
+]
+default_section = "THIRDPARTY"
+sections = [
+    "FUTURE",
+    "STDLIB",
+    "THIRDPARTY",
+    "DASK",
+    "RAPIDS",
+    "FIRSTPARTY",
+    "LOCALFOLDER",
+]
+skip = [
+    ".eggs",
+    ".git",
+    ".hg",
+    ".mypy_cache",
+    ".tox",
+    ".venv",
+    "build",
+    "dist",
+    "__init__.py",
```

Comment:
Again here, is there a reason we need to skip `__init__.py`?

Reply:
Same as before. This was in `setup.cfg` and just got copied over

https://github.com/rapidsai/ucx-py/blob/e6dae58768f9272d8f6c11c4f684bdc09a62b5f9/setup.cfg#L50

Reply:
As before, let's leave it like this for now then.